index.d.ts

```diff
@@ -0,0 +1,20 @@
+/// <reference types="node" />
+
+import { IncomingMessage, ServerResponse } from "node:http";
+
+declare function finalhandler(
+  req: IncomingMessage,
+  res: ServerResponse,
+  options?: finalhandler.Options
+): (err?: any) => void;
+
+declare namespace finalhandler {
+  interface Options {
+    env?: string | undefined;
+    onerror?:
+      | ((err: any, req: IncomingMessage, res: ServerResponse) => void)
+      | undefined;
+  }
+}
```

Comment:
```suggestion
declare function finalhandler(
  req: IncomingMessage,
  res: ServerResponse,
  options?: finalhandler.Options<IncomingMessage, ServerResponse>
): (err?: any) => void;
declare function finalhandler(
  req: Http2ServerRequest,
  res: Http2ServerResponse,
  options?: finalhandler.Options<Http2ServerRequest, Http2ServerResponse>
): (err?: any) => void;

declare namespace finalhandler {
  interface Options<
    TRequest extends IncomingMessage | Http2ServerRequest =
      | IncomingMessage
      | Http2ServerRequest,
    TResponse extends ServerResponse | Http2ServerRequest =
      | ServerResponse
      | Http2ServerRequest
  > {
    env?: string | undefined;
    onerror?: ((err: any, req: TRequest, res: TResponse) => void) | undefined;
  }
}
```
`finalhandler` supports http2. The support was not reflected in `@types/finalhandler`.

Reply:
Hey @RobinTail, Sorry for the ping again. What do you think about http2 support? Would that be the correct way to implement the types?

Reply:
I suggest to detach the feature from copying types as they are.
So that on the first step types could be copied and the corresponding infrastructure and distribution could be established and tested.
And on the next step (another PR) new features/improvements/amendments could be added and you could test that they don't break previously established system, @Phillip9587